pkg/serviceaccount/metadata_test.go

```diff
@@ -0,0 +1,228 @@
+/*
```

Comment:
Is this in the wrong package? Why do you want to add a new package for metadata? I'm not thrilled about exposing extra API functions to split these packages but we aren't at the size where this package is unwieldy or different portions have significantly different clients and dependencies.